packages/caravan-descriptors/caravan-rs/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+pub mod config;
+pub mod utils;
+
+use bdk::bitcoin::Network as BdkNetwork;
+use bdk::descriptor::ExtendedDescriptor as BdkExtendedDescriptor;
+use bdk::KeychainKind::{External, Internal};
+use std::str::FromStr;
+use wasm_bindgen::prelude::*;
+
+#[derive(Debug)]
+#[wasm_bindgen]
+pub struct Network(BdkNetwork);
+
+#[wasm_bindgen]
+impl Network {
+    pub fn from_str(network: &str) -> Result<Network, JsError> {
+        let network = BdkNetwork::from_str(network)?;
+        Ok(Network(network))
+    }
+
+    pub fn to_string(&self) -> String {
+        self.0.to_string()
+    }
+}
+
+#[wasm_bindgen(start)]
+pub fn init() {
+    // initialization code
+}
+
+
+// #[wasm_bindgen]
+// extern "C" {
+//     fn alert(s: &str);
+// }
+
+// #[wasm_bindgen]
+// pub fn init() {
+//     alert("Initialized from Rust!");
+// }
```

Comment:
keep or clean?

Reply:
Oh yeah. Good catch. My adventures in rust and wasm documented 😅

Reply:
[994068e](https://github.com/caravan-bitcoin/caravan/pull/16/commits/994068e300a2404a9f8fd169b78d2973606af278)